### PR TITLE
Enable end to end tests

### PR DESCRIPTION
**Pull request by davodesign84**
To enable end-to-end testing, I developed an adapter that doesn't require extensions to approve transactions (it currently always approves the transaction).

It can either create a random pubkey, or can reconstruct a specific account from the secret key. If we like this solution to enable end to end tests my plan is to also extend it so that you can pre-emptively tell the adapte to reject a transaction so that we can also test for the sad paths. The PR is added to this repo as `jet-v2-app` is not currently working correctly on master.

https://user-images.githubusercontent.com/749029/180613007-65d02abd-741c-4b8b-8394-3dff3e51a814.mov

.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
         "antd": "^4.17.2",
         "bs58": "^4.0.1",
         "chai": "^4.3.6",
+        "e2e-react-adapter": "^0.1.1",
```
Comment (Tristyn) [inline package-lock.json:28]: Could name this @jet-lab/e2e-wallet-adapter

Comment (davodesign84) [reply package-lock.json:28]: Yeah thought about it but I don't have npm access to the org yet so I created it on my mom account :)

File: src/App.tsx [hunk 2]
```diff
@@ -32,20 +33,43 @@ import { NetworkWarningBanner } from './components/NetworkWarningBanner';
...
+      new E2EWalletAdapter(
+        debugWallet.length > 0
+          ? {
+              keypair: Keypair.fromSecretKey(bs58.decode(debugWallet))
+            }
+          : undefined
+      )
+    );
+  }
+  const wallets = useMemo(() => walletArray, []);
```
Comment (Tristyn) [inline src/App.tsx:72]: I think on each render in the background a new E2EWalletAdapter is pushed to the walletArray, walletArray will grow to an unlimited size.
The array creation and the check for devnet could be moved inside useMemo to solve that

Comment (davodesign84) [reply src/App.tsx:72]: Very good shout :)

**Review by Tristyn — COMMENTED** [review]
Very cool

**Comment by Tristyn** [comment]
If you end up only needing a new random wallet every time, I think removing the secret key from the url and calling `Keypair.generate()` would be better. Even though it's just localnet/devnet having no private keys in the app and search history is good practice

**Comment by davodesign84** [comment]
> If you end up only needing a new random wallet every time, I think removing the secret key from the url and calling `Keypair.generate()` would be better. Even though it's just localnet/devnet having no private keys in the app and search history is good practice

I'm not sure about that..for programmatic testing we will need wallets that are preloaded, otherwise we will also have to airdrop everytime.

**Review by Tristyn — APPROVED** [review]

**Pull request merged**